pep-0635.rst

```diff
@@ -0,0 +1,689 @@
+the flipside, if you think of patterns as building on parameters and
+assignment targets, literal patterns are a novel addition (i.e. you would
+not write, e.g., ``(2, a, b) = c`` in iterable unpacking).
+
+Originally, literal patterns came from the idea of expressing unstructured
+singleton objects such as ``None``: instead of requiring that a subject has
+type ``NoneType``, it makes much more sense to directly write ``None``.
+More generally, literal patterns could also be seen as syntactic sugar for
+guards.  Rather than ``case x if x == 0:``, you can simply write ``case 0:``.
+
+Generally, the subject is compared to a literal pattern by means of standard
+equality (``x == y`` in Python syntax).  Consequently, the literal patterns
+``1.0`` and ``1`` match exactly the same set of objects, i.e. ``case 1.0:``
+and ``case 1:`` are fully interchangable.  In principle, ``True`` would also
+match the same set of objects because ``True == 1`` holds.  However, we
+believe that many users would be surprised findings that ``case True:``
+matched the object ``1.0``, resulting in some subtle bugs and convoluted
+work arounds.  We therefore adopted the rule that the three singleton
+objects ``None``, ``False`` and ``True`` match by identity (``x is y`` in
+Python syntax) rather than equality.  Hence, ``case True:`` will match only
+``True`` and nothing else.  Note that ``case 1:`` would still match ``True``,
+though, because the literal pattern ``1`` works by equality and not identity.
+
+Early ideas to induce a hierarchy on numbers so that ``case 1.0`` would
+match both the integer ``1`` and the floating point number ``1.0``, whereas
+``case 1:`` would only match the integer ``1`` very eventually dropped in
+favour of the simpler and consistent rule based on equality.
+
+Recall that literal patterns are *not* expressions, but directly denote a
+specific value or object.  From a syntactical point of view, we have to
+ensure that negative and complex numbers can equally be used as patterns,
+although they are not atomic literal values (i.e. ``-3+4j`` is syntactically
+an expression of the form ``BinOp(UnaryOp('-', 3), '+', 4j)``).
+Interpolated *f*-strings, on the other hand, are not literal values, despite
+their appearance and can therefore not be used as literal patterns.
+
+Literal patterns not only occur as patterns in their own right, but also
+as keys in *mapping patterns*.
+
+
+.. _constant_value_pattern:
+
+Constant Value Patterns
+~~~~~~~~~~~~~~~~~~~~~~~
+
+It is good programming style to use named constants for parametric values or
+to clarify the meaning of particular values.  Clearly, it would be desirable
+to also write ``case (HttpStatus.OK, body):`` rather than
+``case (200, body):``, say.  The main issue that arises here is how to
+discern capture patterns (variables) and constant value patterns.  The
+general discussion surrounding this issue has brought forward a plethora of
+options, which we cannot all fully list here.
```

Comment:
I would at least refer to the idea of marking all stores, so that unqualified names could be used as constant values. And specifically the idea of writing `?var` or `$var`.

Reply:
Totally agreed.  I just was not sure where to put it.  That's why it is not in there, yet.